src/tiled/propertiesdock.cpp

```diff
@@ -293,6 +296,42 @@ void PropertiesDock::renameProperty(const QString &name)
     undoStack->push(new RenameProperty(mMapDocument, mMapDocument->currentObjects(), oldName, name));
 }
 
+void PropertiesDock::showContextMenu(const QPoint& pos)
+{   
+    QtBrowserItem *item = mPropertyBrowser->currentItem();
+    if (!mPropertyBrowser->isCustomPropertyItem(item))
+        return;
+    QPoint globalPos = mPropertyBrowser->mapToGlobal(pos);
+
+    QString name = item->property()->propertyName();
+    Object *object = mMapDocument->currentObject();
+    QVariant value = object->property(name);
+
+    QMenu *contextMenu;
+    contextMenu = new QMenu(mPropertyBrowser);
+    contextMenu->addSection(tr("Convert to:"));
+
+    QList<unsigned int> convertTo;
+    convertTo << QVariant::Bool << QVariant::Color << QVariant::Double << filePathTypeId() << QVariant::Int << QVariant::String;
+    foreach (const unsigned int &toType, convertTo){
+        if (value.canConvert(toType) && (value.type() != toType)) {
+            QAction* action = contextMenu->addAction(typeToName(toType));
+            action->setData(toType);
+        }
+    }
+
+    QAction* selectedItem = contextMenu->exec(globalPos);
+    if (selectedItem) {
+        unsigned int selectedType = selectedItem->data().toInt();
+        if (value.canConvert(selectedType)) {
```

Comment:
This check no longer makes sense since it already happened while adding items to the menu. Actually, rather than storing the selected type, what about storing the converted value directly with the action?